lstchain/tools/lstchain_fit_intensity_scan.py

```diff
@@ -0,0 +1,364 @@
+                        self.selected_runs[chan].append(run)
+                        self.log.info(f"{channel[chan]}: select run {run}, median charge {median_charge:6.1f} ADC\n")
+                else:
+                    raise IOError(f"--> Problem in reading {run}\n")
+
+            # check to have enough selected runs
+            for chan in self.gain_channels:
+                if self.signal[chan] is None:
+                    raise IOError(f"--> Zero runs selected for channel {channel[chan]} \n")
+
+                if self.signal[chan].size < MIN_N_RUNS*constants.N_PIXELS:
+                    raise IOError(f"--> Not enough runs selected for channel {channel[chan]}: {int(self.signal[chan].size/constants.N_PIXELS)} runs \n")
+
+        except ValueError as e:
+            self.log.error(e)
+
+
+    def start(self):
+        '''loop to fit each pixel '''
+
+        # only positive parameters
+        bounds = [0, 200]
+
+        for pix in np.arange(constants.N_PIXELS):
+
+            if pix % 100 == 0 :
+                self.log.debug(f"Pixel {pix}")
+
+            # loop over channel
+            for chan in self.gain_channels:
+
+                # fit parameters initialization
+                p0 = np.array(self.fit_initialization[chan])
+
+                mask = self.unusable_pixels[chan][pix]
+                sig = np.ma.array(self.signal[chan][pix],mask=mask).compressed()
+                var = np.ma.array(self.variance[chan][pix],mask=mask).compressed()
+
+                # skip the pixel if not enough data
+                if sig.shape[0] < MIN_N_RUNS:
+                    self.log.debug(f"Not enough data in pixel {pix} and channel {chan} for the fit ({sig.shape[0]} runs)\n")
+                    self.fit_error[chan,pix] = 1
+                    continue
+
+                # we assume a constant fractional error
+                sigma = self.fractional_variance_error * var
+
+                try:
+                    par, par_cov = curve_fit(quadratic_fit, sig, var, bounds=bounds, sigma=sigma, p0=p0)
+                    self.fit_parameters[chan, pix] = par
+                    self.fit_cov_matrix[chan, pix] = par_cov.reshape(4)
+
+                except Exception as e:
+
+                    self.log.error(e)
+                    self.log.error(f"Error for pixel {pix} and channel {chan}:\n")
+                    self.log.error(f"signal {sig}\n")
+                    self.log.error(f"variance {var}\n")
+
+                    self.fit_error[chan, pix] = 1
+
+    def finish(self):
+        """
+        write fit results in h5 file and the check-plots in pdf file
+        """
+
+        gain = np.ma.array(self.fit_parameters.T[0], mask=self.fit_error.T)
+        quadratic_term = np.ma.array(self.fit_parameters.T[1], mask=self.fit_error.T)
+
+        # give to the badly fitted pixel a median value for the B term
+        median_quadratic_term = np.ma.median(quadratic_term,axis=0)
```

Comment:
I suggest here to add an extra check - if >~5-10% of the pixels have their fit wrong the whole script should end with an error.

Reply:
I prefer to added a plot of the fitted B values over the camera, this will give a clear idea of the pixels that were badly  fit (there are also the text warning)